rule/struct_tag.go

```diff
@@ -73,9 +131,8 @@ func (*StructTagRule) Name() string {
 type lintStructTagRule struct {
 	onFailure      func(lint.Failure)
 	userDefined    map[string][]string // map: key -> []option
```

Comment:
Here also it could be a tagKey


```suggestion
	userDefined    map[tagKey][]string // map: key -> []option
```